### PR TITLE
DOC Add Predictive examples

### PR DESCRIPTION
**Pull request by MarcoGorelli**
xref https://github.com/pyro-ppl/numpyro/pull/1082#issuecomment-872914309

I thought it might be clearer to slightly edit the existing examples in SVI and MCMCKernel, which already have runnable code, and then to refer to them in the Predictive kernel

Here's the part added to Predictive:

image

SVI:

image

MCMCKernel:

image

MCMC.run:

image

---

EDIT
----

some doctest failure...I'll fix this later, the example in `numpyro/infer/util.py` needs to just be code, but without doctest running on it

### CODE REVIEW

File: numpyro/infer/svi.py [hunk 1]
```diff
@@ -101,6 +101,8 @@ class SVI(object):
         >>> svi = SVI(model, guide, optimizer, loss=Trace_ELBO())
         >>> svi_result = svi.run(random.PRNGKey(0), 2000, data)
         >>> params = svi_result.params
+        >>> predictive = Predictive(model, guide=guide, params=params, num_samples=1000)
+        >>> samples = predictive(random.PRNGKey(0), data)
```
Comment (fehiepsi) [inline numpyro/infer/svi.py:105]: This code does not return any useful information unless you set `data=None` (which will give you posterior predictive samples). If you want to get posterior samples, then you can move those codes to the end and add a comment
```python
>>> # get posterior samples
>>> predictive = Predictive(guide, params=params, num_samples=1000)
>>> samples = predictive(random.PRNGKey(1), data)
```

File: numpyro/infer/mcmc.py [hunk 2]
```diff
@@ -64,7 +64,9 @@ class MCMCKernel(ABC):
         >>> kernel = MetropolisHastings(f)
         >>> mcmc = MCMC(kernel, num_warmup=1000, num_samples=1000)
         >>> mcmc.run(random.PRNGKey(0), init_params=jnp.array([1., 2.]))
-        >>> samples = mcmc.get_samples()
+        >>> posterior_samples = mcmc.get_samples()
+        >>> predictive = Predictive(model, posterior_samples=posterior_samples)
+        >>> samples = predictive(rng_key1, *model_args, **model_kwargs)
```
Comment (fehiepsi) [inline numpyro/infer/mcmc.py:69]: I don't think we have a model here. I think you can add a small code snippet to `get_samples` method of `MCMC`.

File: numpyro/infer/mcmc.py [hunk 3]
```diff
@@ -509,6 +511,8 @@ def run(self, rng_key, *args, extra_fields=(), init_params=None, **kwargs):
         """
         Run the MCMC samplers and collect samples.
 
+        See :class:`~numpyro.infer.util.Predictive` for how to use them to collect posterior predictive samples.
```
Comment (fehiepsi) [inline numpyro/infer/mcmc.py:514]: I think `get_samples` is a better place for this. Also maybe changing `them` to `those samples`?

File: numpyro/infer/util.py [hunk 4]
```diff
@@ -743,6 +743,21 @@ class Predictive(object):
           argument is not None, its value should be equal to `num_chains x N`.
 
     :return: dict of samples from the predictive distribution.
+
+    **Example:**
+
+    Given a model, you can sample from the prior predictive:
+
+        >>> predictive = Predictive(model, num_samples=num_samples)
```
Comment (fehiepsi) [inline numpyro/infer/util.py:751]: nit `num_samples=1000`

**Review by fehiepsi — COMMENTED** [review]
Thanks, @MarcoGorelli! I think you can remove `>>> ` to skip doctest.

**Comment by MarcoGorelli** [comment]
Thanks for your feedback! I'll address your comments at the weekend (won't have time this week before then unfortunately)

**Comment by fehiepsi** [comment]
Please take your time. We can always defer doc improvements to later releases. :)

File: numpyro/infer/mcmc.py [hunk 5]
```diff
@@ -593,6 +593,14 @@ def get_samples(self, group_by_chain=False):
...
             but can be any :func:`jaxlib.pytree`, more generally (e.g. when defining a
             `potential_fn` for HMC that takes `list` args).
+
+        **Example:**
+
+        You can then pass those samples to :class:`~numpyro.infer.util.Predictive`:
+
+            posterior_samples = mcmc.get_samples()
+            predictive = Predictive(model, posterior_samples=posterior_samples)
+            samples = predictive(rng_key1, *model_args, **model_kwargs)
```
Comment (fehiepsi) [inline numpyro/infer/mcmc.py:603]: I think you need an empty string after this to make lint pass.

Comment (MarcoGorelli) [reply numpyro/infer/mcmc.py:603]: I think it might be the `*` - will try making it a `r` string

Comment (MarcoGorelli) [reply numpyro/infer/mcmc.py:603]: OK, fixed, it just needed to be put into a code block, so with the previous line ending with `::` rather than `:`

Comment (MarcoGorelli) [reply numpyro/infer/mcmc.py:603]: now it looks like this:

Screenshot 2021-07-11 at 21 17 21

Comment (fehiepsi) [reply numpyro/infer/mcmc.py:603]: yeah, good catch!

**Review by fehiepsi — APPROVED** [review]
Awesome, thanks @MarcoGorelli!

**Pull request merged**